corehq/apps/domain/views.py

```diff
@@ -1238,14 +1238,17 @@ def select_subscription_type_form(self):
 
         subscription_type = None
         subscription = Subscription.get_subscribed_plan_by_domain(self.domain_object)[1]
-        plan = subscription.plan_version.plan
-
-        if subscription.service_type == SubscriptionType.CONTRACTED:
-            subscription_type = "contracted_partner"
-        elif plan.edition == SoftwarePlanEdition.ENTERPRISE:
-            subscription_type = "dimagi_only_enterprise"
-        elif plan.edition == SoftwarePlanEdition.ADVANCED and plan.visibility == SoftwarePlanVisibility.TRIAL:
-            subscription_type = "advanced_extended_trial"
+        try:
```

Comment:
I would rather check whether ```subscription``` is ```None``` than relying on an Exception to be thrown, since that's a normal state to be in (ie there's nothing exceptional about it).